main.tf

```
@@ -71,6 +71,37 @@ resource "aws_dynamodb_table" "this" {
     kms_key_arn = var.server_side_encryption_kms_key_arn
   }
 
+  dynamic "import_table" {
+    for_each = length(var.import_table) > 0 ? [var.import_table] : []
```

```suggestion
    for_each = try([var.import_table], [])
```

or something like that :)

---

Changing to this didn't work as expected so I left it as-is.